client/dashboard/sites/performance/index.tsx

```diff
@@ -1,18 +1,119 @@
-import { Card, CardBody } from '@wordpress/components';
+import { useQuery } from '@tanstack/react-query';
+import { useSearch } from '@tanstack/react-router';
 import { __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import { performanceProfilerPagesQuery, siteQuery } from '../../app/queries';
+import { siteRoute, sitePerformanceRoute } from '../../app/router';
 import { PageHeader } from '../../components/page-header';
 import PageLayout from '../../components/page-layout';
+import { usePerformanceData } from '../hooks/use-performance-data';
+import DeviceTabControls, { ToggleType } from './device-toggle';
+import { PageSelector } from './page-selector';
+import Report from './report';
+import { SubTitle } from './subtitle';
+import type { PerformanceProfilerPage } from '../../data';
+import type { Site } from '../../data/types';
+
+/**
+ * Get the initial page to display based on the page ID.
+ * @param pages - The list of pages to choose from
+ * @param pageId - The ID of the page to display
+ * @returns The initial page to display
+ */
+const getPageFromID = ( pages: PerformanceProfilerPage[] | undefined, pageId: string ) => {
+	return pages?.find( ( page: PerformanceProfilerPage ) => Number( page.id ) === Number( pageId ) );
+};
+
+function SitePerformanceContent( { site }: { site: Site } ) {
+	const { data: pagesData, refetch: refetchPages } = useQuery( {
+		...performanceProfilerPagesQuery( site.ID ),
+		refetchOnWindowFocus: false,
+	} );
+	const { page_id } = useSearch( { from: sitePerformanceRoute.fullPath } );
+	const initialPage = page_id ? getPageFromID( pagesData, page_id ) : pagesData?.[ 0 ];
+	const [ currentPage, setCurrentPage ] = useState( initialPage );
+	const {
+		desktopReport,
+		mobileReport,
+		isLoading: isFetchingReport,
+		isDesktopReportRunning,
+		isMobileReportRunning,
+		isError,
+		isDesktopReportError,
+		isMobileReportError,
+		refetch: refetchReport,
+	} = usePerformanceData( currentPage?.link, currentPage?.wpcom_performance_report_hash );
+	const [ deviceToggle, setDeviceToggle ] = useState< ToggleType >( 'mobile' );
+
+	const handlePageChange = ( pageId: string | null | undefined ) => {
+		const page = getPageFromID( pagesData, pageId || '' );
+
+		setCurrentPage( page );
+	};
+
+	const handleReportRefetch = async () => {
+		await refetchReport();
```

Comment:
Running a sync call to `refetchReport` makes for a predictable code path, however, it does introduce a delay while we are fetching a new token for the test. It's generally pretty fast, but should it be slow, the UI will wait for this to resolve before giving the user any confirmation that we are re-running the test.